app/src/main/java/com/bitchat/android/ui/ChatState.kt

```diff
@@ -24,168 +29,161 @@ data class CommandSuggestion(
+    private val _showAppInfo = MutableStateFlow<Boolean>(false)
+    val showAppInfo: StateFlow<Boolean> = _showAppInfo.asStateFlow()
     
     // Location channels state (for Nostr geohash features)
-    private val _selectedLocationChannel = MutableLiveData<com.bitchat.android.geohash.ChannelID?>(com.bitchat.android.geohash.ChannelID.Mesh)
-    val selectedLocationChannel: LiveData<com.bitchat.android.geohash.ChannelID?> = _selectedLocationChannel
+    private val _selectedLocationChannel = MutableStateFlow<com.bitchat.android.geohash.ChannelID?>(com.bitchat.android.geohash.ChannelID.Mesh)
+    val selectedLocationChannel: StateFlow<com.bitchat.android.geohash.ChannelID?> = _selectedLocationChannel.asStateFlow()
     
-    private val _isTeleported = MutableLiveData<Boolean>(false)
-    val isTeleported: LiveData<Boolean> = _isTeleported
+    private val _isTeleported = MutableStateFlow<Boolean>(false)
+    val isTeleported: StateFlow<Boolean> = _isTeleported.asStateFlow()
     
     // Geohash people state (iOS-compatible)
-    private val _geohashPeople = MutableLiveData<List<GeoPerson>>(emptyList())
-    val geohashPeople: LiveData<List<GeoPerson>> = _geohashPeople
-    // For background thread updates by repositories/handlers in their own scopes
-    val geohashPeopleMutable: MutableLiveData<List<GeoPerson>> get() = _geohashPeople
+    private val _geohashPeople = MutableStateFlow<List<GeoPerson>>(emptyList())
+    val geohashPeople: StateFlow<List<GeoPerson>> = _geohashPeople.asStateFlow()
     
-    
-    private val _teleportedGeo = MutableLiveData<Set<String>>(emptySet())
-    val teleportedGeo: LiveData<Set<String>> = _teleportedGeo
+    private val _teleportedGeo = MutableStateFlow<Set<String>>(emptySet())
+    val teleportedGeo: StateFlow<Set<String>> = _teleportedGeo.asStateFlow()
     
     // Geohash participant counts reactive state (for real-time location channel counts)
-    private val _geohashParticipantCounts = MutableLiveData<Map<String, Int>>(emptyMap())
-    val geohashParticipantCounts: LiveData<Map<String, Int>> = _geohashParticipantCounts
-    
-    // Unread state computed properties
-    val hasUnreadChannels: MediatorLiveData<Boolean> = MediatorLiveData<Boolean>()
-    val hasUnreadPrivateMessages: MediatorLiveData<Boolean> = MediatorLiveData<Boolean>()
+    private val _geohashParticipantCounts = MutableStateFlow<Map<String, Int>>(emptyMap())
+    val geohashParticipantCounts: StateFlow<Map<String, Int>> = _geohashParticipantCounts.asStateFlow()
     
-    init {
-        // Initialize unread state mediators
-        hasUnreadChannels.addSource(_unreadChannelMessages) { unreadMap ->
-            hasUnreadChannels.value = unreadMap.values.any { it > 0 }
+    // Unread state computed properties using Flow.combine
+    val hasUnreadChannels: StateFlow<Boolean> = _unreadChannelMessages
+        .combine(_unreadChannelMessages) { unreadMap, _ ->
+            unreadMap.values.any { it > 0 }
         }
```

Comment:
Using combine is needed when combining data from 2 flows or more, here you're combining the same flow which doesn't make sense, consider using .map{} instead.

```suggestion
        .map { unreadMap -> unreadMap.values.any { it > 0 } }
}
```